docs/getting-started/quickstart-programming.md

```diff
@@ -79,178 +57,232 @@ exports.up = function(knex) {
-We'll do this with a [transactional function](../tutorials/transaction-tutorial.md).
-Copy this code into `src/operations.ts`:
+Now, let's change the app to record every greeting into the `greetings` table. We'll do this with a [transactional function](../tutorials/transaction-tutorial.md). Change your `src/operations.ts` to contain:
 
 ```javascript
-import { TransactionContext, Transaction, HandlerContext, GetApi } from '@dbos-inc/dbos-sdk'
+import { TransactionContext, Transaction, HandlerContext, GetApi } from '@dbos-inc/dbos-sdk';
 import { Knex } from 'knex';
 
+interface GreetingRecord {
+  name: string;
+  note: string;
+}
+
 export class Greetings {
   @Transaction()
-  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
-    await ctxt.client.raw('INSERT INTO greetings (name, note) VALUES (?, ?)', [friend, note]);
-    ctxt.logger.info(`Greeting to ${friend} recorded in the database!`);
+  static async InsertGreeting(ctxt: TransactionContext<Knex>, gr: GreetingRecord) {
+    await ctxt.client('greetings').insert(gr);
+    ctxt.logger.info(`Greeting to ${gr.name} recorded in the database!`);
   }
 
   @GetApi('/greeting/:friend')
   static async Greeting(ctxt: HandlerContext, friend: string) {
     const noteContent = `Thank you for being awesome, ${friend}!`;
-    await ctxt.invoke(Greetings).InsertGreeting(friend, noteContent);
+    await ctxt.invoke(Greetings).InsertGreeting(
+      { name: friend, note: noteContent }
+    );
     return noteContent;
   }
 }
 ```
 
-The key elements of this code are:
-- We use the [`@Transaction`](../api-reference/decorators#transaction) decorator to define a [transactional function](../tutorials/transaction-tutorial.md) (`InsertGreeting`) that accesses the database from a managed client.
-- Inside `InsertGreeting`, we insert a row in the database with `ctxt.client.raw()`.
-- We add a line to `Greeting` invoking `InsertGreeting`.
+Here we define a `GreetingRecord` interface to match a row of data in our `greetings` table. We then define a `@Transaction` called `InsertGreeting` that puts a new `GreetingRecord` into the `greetings` table. Finally, we add a line to the GET API function `Greeting` to invoke `InsertGreeting` with the provided `name` and a welcoming `note`.
+
+Stop your app with CTRL+C. Rebuild with `npm run build` and start with `npx dbos start`. Make a few visits to the greeting URL in your browser, i.e. http://localhost:3000/greeting/Mike. With every new visit, the app should print this to the console:
+```
+[info]: Greeting to Mike recorded in the database! 
+```
+
+You can then query your Postgres database to see the greeting records. It may look something like this:
+```
+psql -h localhost -U postgres -d hello -c "select * from greetings"; 
+```
+The database name `hello` above will vary with your application name. Your `host`, `port` or `username` may also be differnt depending on how you configured Postgres in the Quickstart.
```

Comment:
I think we want to drop this block. It breaks the flow of this programming guide, and we want to really focus on programming here.

Reply:
...we could... but when I was personally going thru the guide. I ran the postgres thru docker. And I wasn't sure how exactly it was set up... and it took me like 30 minutes to discover that I can do this -and I found it very useful....

Reply:
That's fair, but it may belong to the quickstart guide where we introduce Postgres setup.

Reply:
Yep, moved this section to quickstart in the latest commit; have a look